### PR TITLE
Add end-to-end gameplay test

### PR DESCRIPTION
**Pull request by Brainicism**

### CODE REVIEW

File: src/test/end-to-end-tests/test_suites/play_test.ts [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+                parsedGameOptions?: ParsedGameOptionValues,
+                client?: Eris.Client,
+            ) =>
+                parsedGameOptions!["limit set top"]!.value.startsWith("6 /") &&
+                parsedGameOptions!["limit set top"]!.updated,
+            expectedResponseType: KmqResponseType.GAME_OPTIONS_RESPONSE,
+            requiresVoiceConnection: true,
+        },
+        {
+            command: ",shuffle popularity", // deterministic ordering so we can figure out how to guess
```
Comment (taahamahdi) [inline src/test/end-to-end-tests/test_suites/play_test.ts:37]: Will this ordering break once one of these six songs changes rank? Should we query the DB to get the song/artist names instead

File: src/test/end-to-end-tests/test_suites/play_test.ts [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+                title: string,
+                description: string,
+                parsedGameOptions?: ParsedGameOptionValues,
+                client?: Eris.Client,
+            ) => true,
+            expectedResponseType: KmqResponseType.RAW,
+            requiresVoiceConnection: true,
+        },
+        {
+            command: "gangnma style", // gangnam style - psy, lost guess streak, unique songs reset, allow typo
```
Comment (taahamahdi) [inline src/test/end-to-end-tests/test_suites/play_test.ts:175]: Do we already have a command that sets it to typingtypos elsewhere? Is it a part of the other test file?

Comment (Brainicism) [reply src/test/end-to-end-tests/test_suites/play_test.ts:175]: It's the default

File: src/test/end-to-end-tests/test_suites/play_test.ts (hunk 1 above)
Comment (Brainicism) [reply src/test/end-to-end-tests/test_suites/play_test.ts:37]: Yes but would rather keep the test runner agnostic of anything internal, it should be as close to a regular player as possible. 

Many of our tests break when the database changes in certain ways. We use a cached version of the database so it doesn't break unless we update it, and fixing things doesn't take much effort

Comment (taahamahdi) [reply src/test/end-to-end-tests/test_suites/play_test.ts:37]: Good point, let's leave it as is then

**Review by taahamahdi — APPROVED** [review]

**Review by taahamahdi — APPROVED** [review]

**Pull request merged**